tools/filter_cells_and_genes/filter_cells_and_genes.xml

```diff
@@ -0,0 +1,270 @@
+         <param name="file_sep" value="tab"/>
+         <param name="file_header" value="TRUE"/>
+         <param name="is_mito" value="TRUE"/>
+         <param name="header_mito" value="MT-"/>
+         <param name="package" value="Seurat"/>
+         <param name="min_genes" value="10"/>
+         <param name="min_cells" value="3"/>
+         <output name="output_plot" file="counts_with_mito_Seurat_filter_plots.pdf" ftype="pdf"/>
+         <output name="output_matrix" file="counts_with_mito_Seurat_filtered.tab" ftype="tabular"/>
+    </test> 
+   </tests>
+   <help>
+
+**What it does**
+
+Filter_cells_and_genes takes a tabular table of raw read counts, each column corresponding to a cell and each row corresponding to a
+gene, and returns (1) a PDF with quality controls plots (like the distribution of library sizes or expressed genes by cell, frequency of 
+expression or the most expressed genes) and (2) a filtered table of counts.
+
+You can filter out low quality cells from your raw expression table :
+
+- Based on expression level of mitochondrial genes :
+
+    - Using R package scater: remove cells that detect mitochondrial genes 3 times higher than the median of all cells
+    - Using R package Seurat: if cells expressed more than 20% of mitochondrial genes, they are removed.
+
+- Using R package scater : search cells with low library size and few detected genes based on the deviation from the median of all cel$
+
+- Using R package Seurat : Remove all cells that detected at least n genes.
+
+Also you can filter out low quality genes from your raw expression table :
+
+- Using R package scater :
+
+    - Remove all genes that aren't detected in at least n cells
+    - Filter out genes based on its average count (which is adjust for library size between cells). It is more stringent.
+
+- Using R package Seurat : Remove all genes that aren't detected in at least n cells.
+
+
+**Inputs**
+
+The following table gives an example of the tabular input that has to be provided to the tool:
+
+
+<![CDATA[
+
+AGTCAGACTAGAGAAGTCCAGATTTCAC	AGTCTACTTGCATG	AGTGTTCTATAAGG	AGTGTTCTCACTTT	AGTTCTTGAAGCCT	ATAAACACAGTGCT	ATAACCCTGTTGGT	ATAATCGAGCTGAT	ATACCACTCGTACA  
+
+MALAT1	34	78	45	86	12	46	72	76	33	19  
+
+TMSB4X	32	33	109	27	46	25	31	43	11	46  
+
+B2M	53	64	93	21	10	40	33	33	18	15  
+
+RPL10	35	19	36	24	35	20	16	24	32	32  
+
+RPL13	26	23	37	17	28	12	32	24	18	56  
+
+RPL13A	16	32	38	24	21	10	18	30	19	25  
+
+FTL	2	4	175	6	101	6	31	2	1	54  
+
+RPS2	17	20	32	8	35	10	23	20	12	24  
+
+RPS6	57	20	28	15	25	14	24	29	13	17  
+
+FTH1	6	7	151	2	22	5	9	7	4	45  
+
+RPS18	9	25	19	16	17	8	21	26	9	19  
+
+RPL11	19	15	30	14	21	7	13	15	8	64  
+
+RPL32	14	16	30	15	30	10	15	17	13	22  
+
+RPS3	9	19	12	12	12	11	16	18	4	11  
+
+RPL19	16	16	27	10	70	11	16	14	11	17  
+
+MT-ND1	2	0	5	1	6	4	6	3	2	5  
+
+MT-ND2	4	4	15	2	0	1	3	9	0	8  
+
+MT-CO1	11	3	52	10	9	14	11	9	9	22  
+
+MT-CO2	4	5	35	8	5	2	7	10	2	7  
+
+MT-ATP8	0	0	0	0	0	0	0	0	0	0  
+
+
```

Comment:
** Outputs **

And here we are. Sorry Lea but you will  have to explain every possible output, especially the pdf. This is the (nice) job: explaining the outputs is just *Science* - and make all difference with bad "have to guess" tools !

Reply:
@bellenger-l The value of a tool often stands in its documentation